### ISSUE TITLE
Sometimes, when stopping download, a ConcurrentModificationException is thrown

### ISSUE THREAD

**Issue by Pokechu22** [Pokechu22/WorldDownloader#13]
Crash report: 

```
[20:38:49] [Client thread/FATAL]: Unreported exception thrown!
java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextEntry(HashMap.java:922) ~[?:1.7.0_75]
	at java.util.HashMap$KeyIterator.next(HashMap.java:956) ~[?:1.7.0_75]
	at com.google.common.collect.AbstractMapBasedMultimap$WrappedCollection$WrappedIterator.next(AbstractMapBasedMultimap.java:486) ~[guava-17.0.jar:?]
	at net.minecraft.util.ClassInheratanceMultiMap$2.computeNext(ClassInheratanceMultiMap.java:116) ~[bin/:?]
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143) ~[guava-17.0.jar:?]
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138) ~[guava-17.0.jar:?]
	at net.minecraft.client.renderer.RenderGlobal.func_180446_a(RenderGlobal.java:625) ~[bin/:?]
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1405) ~[bin/:?]
	at net.minecraft.client.renderer.EntityRenderer.renderWorld(EntityRenderer.java:1304) ~[bin/:?]
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1117) ~[bin/:?]
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1104) ~[bin/:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:419) [bin/:?]
	at net.minecraft.client.main.Main.main(Main.java:150) [bin/:?]
	at Start.main(Start.java:14) [bin/:?]
---- Minecraft Crash Report ----
// I blame Dinnerbone.

Time: 10/16/15 8:38 PM
Description: Unexpected error

java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextEntry(HashMap.java:922)
	at java.util.HashMap$KeyIterator.next(HashMap.java:956)
	at com.google.common.collect.AbstractMapBasedMultimap$WrappedCollection$WrappedIterator.next(AbstractMapBasedMultimap.java:486)
	at net.minecraft.util.ClassInheratanceMultiMap$2.computeNext(ClassInheratanceMultiMap.java:116)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at net.minecraft.client.renderer.RenderGlobal.func_180446_a(RenderGlobal.java:625)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1405)
	at net.minecraft.client.renderer.EntityRenderer.renderWorld(EntityRenderer.java:1304)
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1117)
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1104)
	at net.minecraft.client.Minecraft.run(Minecraft.java:419)
	at net.minecraft.client.main.Main.main(Main.java:150)
	at Start.main(Start.java:14)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at java.util.HashMap$HashIterator.nextEntry(HashMap.java:922)
	at java.util.HashMap$KeyIterator.next(HashMap.java:956)
	at com.google.common.collect.AbstractMapBasedMultimap$WrappedCollection$WrappedIterator.next(AbstractMapBasedMultimap.java:486)
	at net.minecraft.util.ClassInheratanceMultiMap$2.computeNext(ClassInheratanceMultiMap.java:116)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at net.minecraft.client.renderer.RenderGlobal.func_180446_a(RenderGlobal.java:625)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1405)

-- Affected level --
Details:
	Level name: MpServer
	All players: 42 total; [EntityPlayerSP['pokechu22'/48992513, l='MpServer', x=0.69, y=72.00, z=5.41], EntityOtherPlayerMP['Aerick'/48693613, l='MpServer', x=0.56, y=74.02, z=0.75], EntityOtherPlayerMP['JamesVGC'/26324735, l='MpServer', x=-11.31, y=71.06, z=28.00], EntityOtherPlayerMP['Ecstasty'/48199350, l='MpServer', x=-3.22, y=71.50, z=9.22], EntityOtherPlayerMP['Vinilha10'/46673995, l='MpServer', x=-2.91, y=71.00, z=17.38], EntityOtherPlayerMP['HeSoPanda'/47171055, l='MpServer', x=10.88, y=78.00, z=-11.88], EntityOtherPlayerMP['butterwarrior64'/47040295, l='MpServer', x=-89.53, y=67.00, z=36.56], EntityOtherPlayerMP['AkramPlays'/44227968, l='MpServer', x=10.88, y=78.00, z=-11.88], EntityOtherPlayerMP['NerdsRCool'/35173036, l='MpServer', x=11.94, y=72.00, z=-11.72], EntityOtherPlayerMP['secretblast'/48974661, l='MpServer', x=0.50, y=74.00, z=-1.00], EntityOtherPlayerMP['ArtZ_AJ'/24994514, l='MpServer', x=39.66, y=71.00, z=-11.72], EntityOtherPlayerMP['JOOSEY_JOOSE'/48854855, l='MpServer', x=51.16, y=-57.20, z=60.78], EntityOtherPlayerMP['K838'/35449426, l='MpServer', x=-17.56, y=71.02, z=-1.53], EntityOtherPlayerMP['Hiro_zg'/37962802, l='MpServer', x=-25.81, y=71.08, z=-7.09], EntityOtherPlayerMP['Blue_Hawkeye'/35694997, l='MpServer', x=8.97, y=82.00, z=-18.69], EntityOtherPlayerMP['GentleCaab'/41755966, l='MpServer', x=4.19, y=71.00, z=-31.78], EntityOtherPlayerMP['SenorDuckXDDD'/37626033, l='MpServer', x=8.97, y=82.00, z=-18.69], EntityOtherPlayerMP['ImmortalElla'/37551346, l='MpServer', x=-36.47, y=72.00, z=-3.94], EntityOtherPlayerMP['fiercepierce772'/40772154, l='MpServer', x=-36.09, y=71.02, z=-25.84], EntityOtherPlayerMP['MarcoSharko77'/19013048, l='MpServer', x=5.69, y=72.00, z=-33.72], EntityOtherPlayerMP['Harper9753'/48776045, l='MpServer', x=10.38, y=72.00, z=-5.91], EntityOtherPlayerMP['Brogan366'/49022771, l='MpServer', x=11.59, y=70.75, z=-3.88], EntityOtherPlayerMP['spear5585818'/49063222, l='MpServer', x=-7.97, y=176.00, z=29.16], EntityOtherPlayerMP['cedrian80'/48945830, l='MpServer', x=35.69, y=71.02, z=8.69], EntityOtherPlayerMP['ASAHI32'/49145607, l='MpServer', x=38.06, y=66.47, z=37.44], EntityOtherPlayerMP['Purple_JOEY'/49146719, l='MpServer', x=1.72, y=176.00, z=0.28], EntityOtherPlayerMP['_TotalHard_'/49129917, l='MpServer', x=-5.50, y=176.00, z=-4.69], EntityOtherPlayerMP['VirusPlayzMC'/49169377, l='MpServer', x=-28.78, y=74.00, z=36.97], EntityOtherPlayerMP['Kaitycatt'/49205745, l='MpServer', x=20.94, y=72.00, z=40.84], EntityOtherPlayerMP['diamondsword4112'/49136567, l='MpServer', x=-4.34, y=72.00, z=5.66], EntityOtherPlayerMP['xXenchanted68Xx'/49221708, l='MpServer', x=-7.91, y=72.00, z=9.72], EntityOtherPlayerMP['Coolin40'/49221709, l='MpServer', x=-1.69, y=72.88, z=4.56], EntityOtherPlayerMP['live3d'/49205989, l='MpServer', x=0.47, y=72.02, z=3.31], EntityOtherPlayerMP['ChaosBlade987'/49232335, l='MpServer', x=3.97, y=72.02, z=6.94], EntityOtherPlayerMP['FistoFury'/49231439, l='MpServer', x=-7.34, y=72.00, z=7.34], EntityOtherPlayerMP['LeePete'/49221917, l='MpServer', x=-35.47, y=71.02, z=-27.06], EntityOtherPlayerMP['Obermensch'/49242816, l='MpServer', x=42.56, y=71.31, z=-8.38], EntityOtherPlayerMP['racen55'/49248303, l='MpServer', x=-2.78, y=71.06, z=14.97], EntityOtherPlayerMP['Jake54722'/49251921, l='MpServer', x=8.84, y=71.50, z=-3.69], EntityOtherPlayerMP['crazyjaker'/49251351, l='MpServer', x=0.56, y=71.02, z=15.00], EntityOtherPlayerMP['darknight07'/49251816, l='MpServer', x=-28.41, y=72.50, z=23.09], EntityOtherPlayerMP['Nesa'/49242817, l='MpServer', x=-2.22, y=72.00, z=1.91]]
	Chunk stats: MultiplayerChunkCache: 169, 169
	Level seed: 0
	Level generator: ID 01 - flat, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: -447.00,4.00,2352.00 - World: (-447,4,2352), Chunk: (at 1,0,0 in -28,147; contains blocks -448,0,2352 to -433,255,2367), Region: (-1,4; contains chunks -32,128 to -1,159, blocks -512,0,2048 to -1,255,2559)
	Level time: 15192276 game time, 132522787056 day time
	Level dimension: 0
	Level storage version: 0x04ABD - Anvil
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: ~~ERROR~~ ClassCastException: null
	Retry entities: 0 total; []
	Server brand: BungeeCord (git:BungeeCord-Bootstrap:1.8-SNAPSHOT:"fa828eb":1059) <- Spigot
	Server type: Non-integrated multiplayer server

-- System Details --
Details:
	Minecraft Version: 1.8
	Operating System: Windows 8.1 (amd64) version 6.3
	Java Version: 1.7.0_75, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 380768000 bytes (363 MB) / 618659840 bytes (590 MB) up to 1308622848 bytes (1248 MB)
	JVM Flags: 1 total; -Xbootclasspath:C:\Program Files\Java\jdk1.7.0_75\jre\lib\resources.jar;C:\Program Files\Java\jdk1.7.0_75\jre\lib\rt.jar;C:\Program Files\Java\jdk1.7.0_75\jre\lib\jsse.jar;C:\Program Files\Java\jdk1.7.0_75\jre\lib\jce.jar;C:\Program Files\Java\jdk1.7.0_75\jre\lib\charsets.jar;C:\Program Files\Java\jdk1.7.0_75\jre\lib\jfr.jar;C:\Program Files\Java\jdk1.7.0_75\lib\tools.jar
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: mcp
	LWJGL: 2.9.1
	OpenGL: P????L??	??^??z??? GL version 4.4.13374 Compatibility Profile Context 15.200.1012.2, ATI Technologies Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: No
	Is Modded: Definitely; Client brand changed to 'worlddownloader1.8d-coremod'
	Type: Client (map_client.txt)
	Resource Packs: []
	Current Language: English (US)
	Profiler Position: N/A (disabled)
```

Seems to be something with entity processing messing.  This has been around for a while and is hard to reproduce consistently unfortunately.

**Comment by Pokechu22** [comment]
Pretty sure that the fix I made in 30398f7 has solved this.  It's an odd solution but it seems to be good.

... changing the return value based off of the stacktrace.  Ugh :/

**Comment by Pokechu22** [comment]
Fixed in 1.8d and all related builds (which have now been released).
